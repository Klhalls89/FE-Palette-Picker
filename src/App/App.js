import React, { Component } from 'react';

import './App.css';

class App extends Component {
  constructor(){
    super()
    this.state = {
      colors: []
    }
  }

  render() {
    return (
      <div className="App">
        <h1 className="Logo">Palette Picker</h1>
      </div>
    )
  }
}

export default App;
