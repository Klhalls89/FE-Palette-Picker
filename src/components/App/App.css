* {
  box-sizing: border-box;
  font-family: 'Open Sans', sans-serif;
}

.App {
  text-align: center;
  font-family: 'Open Sans', sans-serif;
}

.logo {
  font-size: 50px;
}

.main-colors {
  display: flex;
  justify-content: space-between;
  margin: 1%;
}

.main-colors div {
  display: flex;
  flex-direction: column;
  height: 400px;
  justify-content: flex-end;
  margin: 0.5%;
  padding: 1%;
  width: 300px;
}

.fas {
  font-size: 26px;
  padding-bottom: 15px;
}

.fas:hover {
  color: #f7f7f7;
  transform: scale(1.2);
  cursor: pointer;
}

select,
input {
  width: 300px;
  height: 40px;
}

input {
  border-radius: 1px;
  border: solid #C3C3C3 1px;
}

select {
  border: none;
}

button {
  background-color: #C3C3C3;
  width: 300px;
  height: 40px;
  font-size: 18px;
  margin: 25px 0;
}

#gen-colors {
  border: none;
}

#gen-colors:hover {
  cursor: pointer;
}
