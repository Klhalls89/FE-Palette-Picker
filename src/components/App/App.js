/* eslint-disable no-console */
/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import ProjectsContainer from '../ProjectsContainer/ProjectsContainer';
import SubmitProject from '../SubmitProject/SubmitProject';
import { fetchInfo, newFolder } from '../../Utility/ApiCalls';
import PaletteForm from '../PaletteForm/PaletteForm';

class App extends Component {
  constructor() {
    super();
    this.state = {
      colors: 
      [
        { color: '', isLocked: false },
        { color: '', isLocked: false },
        { color: '', isLocked: false },
        { color: '', isLocked: false },
        { color: '', isLocked: false } 
      ],
      projects: [],
      folder: {},
      allPalettes: []
    }; 
  }

  componentDidMount() {
    this.makeColors();
    this.getProjects();
    this.getPalettes();
  }

  getProjects = async () => {
    const projects = await fetchInfo('projects');
    this.setState({projects});
  }

  getPalettes = async () => {
    const allPalettes = await fetchInfo('palettes');
    this.setState({allPalettes});
  }

  editPalette = (colors) => {
    console.log(colors)
  }

  makeColors = () => {
    if (!this.state.colors.length) {
      let colors = [];

      for (let i = 0; i < 5; i++) {
        let color = this.genHex();
        colors.push(color);
      }
      this.setState({colors});

    } else {
      let newColors = this.state.colors.map(color => {
        if (color.isLocked == false) {
          return color = this.genHex();
        } else {
          return color;
        }
      });

      this.setState({colors: newColors});
    }
  };

  genHex = () => {
    let hexables = 
      [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'A', 'B', 'C', 'D', 'E', 'F' ];
    let shuffledHexables = hexables.sort(() => Math.random() - 0.5);
    let newColor = shuffledHexables.reduce((finalColor, hexable) => {
      
      finalColor += hexable;
      return finalColor.slice(0, 6);
    }, '');

    return {
      color: `#${newColor}`,
      isLocked: false
    };
  };

  handleColor = (location) => {
    const { colors } = this.state;
    const newLock = [...colors ];

    newLock[`${location}`].isLocked = !newLock[`${location}`].isLocked;
    this.setState({ colors: newLock });
  }

  handleProject = (section, name) => {
    const {colors } = this.state;
    let preFolder = {};

    if (section === 'project') {
      const postFolder = {...this.state.folder, project_title: name};
      newFolder(postFolder);
    } else {
      for (let i = 1; i < colors.length + 1; i++) {
        const cleanColor = colors[i - 1].color.replace('#', '');
        preFolder[`color_${[i]}`] = cleanColor;
      }
      preFolder.palette_title = name;
      this.setState({folder: preFolder});
    }
  }


  render() {
    const { colors, projects, allPalettes } = this.state;
    return (
      <div className="App">
        <h1 className="logo">Gen Hex</h1>
        <section className="main-colors">
          <div style={{backgroundColor: colors[0].color}} className="color-1">
            <p>{colors[0].color}</p>
            <i className={`fas ${ (colors[0].isLocked) ? 
              'fa-lock' : 'fa-unlock'}`} onClick={() => this.handleColor(0)}>
            </i>
          </div>
          <div style={{backgroundColor: colors[1].color}} className="color-2">
            <p>{colors[1].color}</p>
            <i className={`fas ${ (colors[1].isLocked) ? 
              'fa-lock' : 'fa-unlock'}`} onClick={() => this.handleColor(1)}>
            </i>        
          </div>
          <div style={{backgroundColor: colors[2].color}} className="color-3">
            <p>{colors[2].color}</p>
            <i className={`fas ${ (colors[2].isLocked) ? 
              'fa-lock' : 'fa-unlock'}`} onClick={() => this.handleColor(2)}>
            </i>
          </div>
          <div style={{backgroundColor: colors[3].color}} className="color-4">
            <p>{colors[3].color}</p>
            <i className={`fas ${ (colors[3].isLocked) ? 
              'fa-lock' : 'fa-unlock'}`} onClick={() => this.handleColor(3)}>
            </i>
          </div>
          <div style={{backgroundColor: colors[4].color}} className="color-5">
            <p>{colors[4].color}</p>
            <i className={`fas ${ (colors[4].isLocked) ? 
              'fa-lock' : 'fa-unlock'}`} onClick={() => this.handleColor(4)}>
            </i>
          </div>
        </section>
        <button onClick={() => this.makeColors()} className="gen-colors">
          Generate Colors
        </button>
        { projects.length && <PaletteForm 
          projects={this.state.projects} 
          handleProject={this.handleProject}/>
        }
        <SubmitProject
          handleProject={this.handleProject}/>
        { allPalettes.length > 0 && 
          <ProjectsContainer projects={projects} 
            allPalettes={allPalettes} editPalette={this.editPalette}/>
        } 
      </div>
    );
  }
}

export default App;
