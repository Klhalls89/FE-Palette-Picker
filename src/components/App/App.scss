* {
  box-sizing: border-box;
}

.App {
  text-align: center;
  font-family: $main-font;
}

.logo {
  font-family: $accent-font;
  font-size: 50px;
}

.main-colors {
  display: flex;
  justify-content: space-between;
  margin: 1%;

  div {
    display: flex;
    flex-direction: column;
    height: 400px;
    justify-content: flex-end;
    margin: .5%;
    padding: 1%;
    width: 300px;
  }
}

select,
input {
  width: 300px;
  height: 40px;
}

input {
  border-radius: 1px;
  border: solid #C3C3C3 1px;
}

select {
  border: none;
}

button {
  background-color: #C3C3C3;
  width: 300px;
  height: 40px;
  font-size: 18px;
  margin: 25px 0;
}


