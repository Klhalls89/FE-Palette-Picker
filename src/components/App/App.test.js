import React from 'react';
import App from './App';
import {shallow } from 'enzyme'

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentDidMount', () => {
    it.skip('should call makeColors', () => {
      const makeColors = jest.fn()
      expect(makeColors).toHaveBeenCalledTimes(1)
    })
    it.skip('should call getProjects', () => {
      let getProjects = jest.fn()
      expect(getProjects).toHaveBeenCalledTimes(1)
    })
  })

  describe('makeColors', () => {
    it.skip('should call genHex the passed amount of times', () => {
      //setup
      const genHex = jest.fn()
      wrapper.instance.makeColors()
      expect(genHex).toHaveBeenCalled()
    })
  })

  describe('genHex', () => {
    it('should output a hex value when called', () => {

    })
  })

  describe('render', () => {
    it('should match the snapshot', () => {

    })
  })
})


