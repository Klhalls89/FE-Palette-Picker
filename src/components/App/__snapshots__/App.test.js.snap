// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="logo"
  >
    Gen Hex
  </h1>
  <section
    className="main-colors"
  >
    <div
      className="color-1"
      style={
        Object {
          "backgroundColor": "#2FEB7C",
        }
      }
    >
      <p>
        #2FEB7C
      </p>
      <i
        className="fas fa-unlock"
        onClick={[Function]}
      />
    </div>
    <div
      className="color-2"
      style={
        Object {
          "backgroundColor": "#E84316",
        }
      }
    >
      <p>
        #E84316
      </p>
      <i
        className="fas fa-unlock"
        onClick={[Function]}
      />
    </div>
    <div
      className="color-3"
      style={
        Object {
          "backgroundColor": "#83D2B6",
        }
      }
    >
      <p>
        #83D2B6
      </p>
      <i
        className="fas fa-unlock"
        onClick={[Function]}
      />
    </div>
    <div
      className="color-4"
      style={
        Object {
          "backgroundColor": "#09C524",
        }
      }
    >
      <p>
        #09C524
      </p>
      <i
        className="fas fa-unlock"
        onClick={[Function]}
      />
    </div>
    <div
      className="color-5"
      style={
        Object {
          "backgroundColor": "#E031B5",
        }
      }
    >
      <p>
        #E031B5
      </p>
      <i
        className="fas fa-unlock"
        onClick={[Function]}
      />
    </div>
  </section>
  <button
    id="gen-colors"
    onClick={[Function]}
  >
    Generate Colors
  </button>
  0
  <SubmitProject
    edit={false}
    folderToEdit={Object {}}
    handleProject={[Function]}
    projects={Array []}
  />
</div>
`;
